### ISSUE TITLE
BigQuery: refactor `bigquery_query_params_*` code samples to pass `query_parameters` to `QueryJobConfig` constructor 

### ISSUE THREAD

**Issue by tswast** [googleapis/google-cloud-python#9596]
**Is your feature request related to a problem? Please describe.**

The code samples for using query parameters are a bit hard to follow because they create several temporary variables.

https://github.com/googleapis/google-cloud-python/blob/518931bb306ed7cb78ce76323fbde289fbc15338/bigquery/docs/snippets.py#L1868-L1873

The code samples are still in `snippets.py`. Ideally these samples would be moved to the `samples/` directory.

The code samples supply a `location` argument for queries, even though this can now be autodetected for queries in all regions.

 **Describe the solution you'd like**

- [ ] Move any `bigquery_query_params_...` samples to the `samples/` directory.
- [ ] Remove the `location` argument from the `query()` method in the moved code samples
- [ ] Use the `query_parameters` argument in the constructor to `QueryJobConfig` (added in https://github.com/googleapis/google-cloud-python/pull/6397).

```python
job_config = bigquery.QueryJobConfig(
    query_parameters=[
        bigquery.ScalarQueryParameter("corpus", "STRING", "romeoandjuliet"),
        bigquery.ScalarQueryParameter("min_word_count", "INT64", 250),
    ],
)
```

 **Describe alternatives you've considered**

N/A

 **Additional context**

Reported by customer in internal issue 143862711.

**Comment by Emar-Kar** [comment]
Since I’m already working with the `snippets` refactoring #9495, I assume this should be assigned to me as well.
